warning from gcc 12.1

Newest g++ gives a warning about use of deprecated feature
```
In file included from src/slha_io.cpp:27:
slhaea/slhaea.h:1528:36: warning: ‘template<class _Arg, class _Result> struct std::unary_function’ is deprecated [-Wdeprecated-declarations]
 1528 |   struct key_matches : public std::unary_function<value_type, bool>
      |                                    ^~~~~~~~~~~~~~
In file included from /usr/include/c++/12/string:48,
                 from src/slha_format.hpp:22,
                 from src/slha_io.hpp:22,
                 from src/slha_io.cpp:19:
/usr/include/c++/12/bits/stl_function.h:117:12: note: declared here
  117 |     struct unary_function
      |            ^~~~~~~~~~~~~~
slhaea/slhaea.h:2354:36: warning: ‘template<class _Arg, class _Result> struct std::unary_function’ is deprecated [-Wdeprecated-declarations]
 2354 |   struct key_matches : public std::unary_function<value_type, bool>
      |                                    ^~~~~~~~~~~~~~
/usr/include/c++/12/bits/stl_function.h:117:12: note: declared here
  117 |     struct unary_function
      |            ^~~~~~~~~~~~~~
slhaea/slhaea.h:2375:46: warning: ‘template<class _Arg, class _Result> struct std::unary_function’ is deprecated [-Wdeprecated-declarations]
 2375 |   struct key_matches_block_def : public std::unary_function<value_type, bool>
      |                                              ^~~~~~~~~~~~~~
/usr/include/c++/12/bits/stl_function.h:117:12: note: declared here
  117 |     struct unary_function
      |            ^~~~~~~~~~~~~~
```
This is by no mean critical but I thought you might want to know about that.

--- a/slhaea.h
+++ b/slhaea.h
@@ -1525,7 +1525,7 @@ public:
   { std::for_each(begin(), end(), MEM_FN(&value_type::uncomment)); }
 
   /** Unary predicate that checks if a provided key matches a Line. */
-  struct key_matches : public std::unary_function<value_type, bool>
+  struct key_matches
   {
     explicit
     key_matches(const key_type& key) : key_(key) {}
@@ -2351,7 +2351,7 @@ public:
    * Unary predicate that checks if a provided name matches the name
    * of a Block.
    */
-  struct key_matches : public std::unary_function<value_type, bool>
+  struct key_matches
   {
     explicit
     key_matches(const key_type& blockName) : name_(blockName) {}
@@ -2372,7 +2372,7 @@ public:
    * Unary predicate that checks if a provided key matches the block
    * definition of a Block.
    */
-  struct key_matches_block_def : public std::unary_function<value_type, bool>
+  struct key_matches_block_def
   {
     explicit
     key_matches_block_def(const value_type::key_type& key)

